src/site/content/en/blog/service-worker-mindset/index.md

```diff
@@ -0,0 +1,204 @@
+tags:
+  - post # post is a required tag for the article to show up in the blog.
+  - service-worker
+  - PWA
+  - offline
+  - caching
+---
+
+Service Workers are powerful and absolutely worth mastering. They'll enable you to deliver an entirely new level of experience to your users. But Service Workers are unlike anything most of us web devs are used to. They come with a steep learning curve and a handful of snags you've got to watch out for.
+
+The best way to ensure success with Service Workers is to have the right mindset when working with them. In this post we'll wrap our brains around the paradoxical traits that make Service Workers both tricky and awesome.
+
+## The Same, but Different
+
+While coding your Service Worker, many things will feel familiar. You get to use your favorite new JavaScript language features. You listen to lifecycle events just like with UI events. You manage control flow with promises like you're used to.
+
+But other Service Worker behavior causes you to scratch your head in confusion. Especially when you refresh the page and don't see your code changes applied.
+
+### A New Layer
+
+Normally when building a site you have just two layers to think about: the client and the server. Service Worker is a brand new layer that sits in the middle.
+
+![Middle Layer](./middle-layer.jpg)
+
+Think of your Service Worker as a sort of _browser extension_ — one that your site can install in your users' browser. Once installed, the Service Worker _extends_ the browser for your site with a powerful middle layer. This Service Worker layer can intercept and handle all of the requests your site makes.
+
+The Service Worker layer has its **own lifecycle** independent of the browser tab. A simple page refresh isn't enough to update it. Just like you wouldn't expect a page refresh to update code deployed on a server. Each layer has its own unique rules for updating.
+
+In the [Service Workies](https://serviceworkies.com) game we cover the many details of the Service Worker lifecycle and give you a ton of practice working with it.
+
+![service worker lifecycle update](./update-lifecycle.gif)
+
+{% Aside 'note' %}
+Think of your Service Worker as a new middle layer, with its own lifecycle and methods for updating.
+{% endAside %}
+
+## Powerful, but Limited
+
+Having a Service Worker on your site gives you incredible benefits. Your site can:
+
+- work flawlessly even when the user is offline
+- gain massive performance improvements through [caching](https://developer.mozilla.org/en-US/docs/Web/API/Cache)
+- use [push notifications](https://developers.google.com/web/fundamentals/push-notifications/)
+- be installed as a [PWA](https://developers.google.com/web/progressive-web-apps/)
+
+With as much as Server Workers can do, they are limited by design. They can't do anything synchronous or in the same thread as your site. So that means no access to:
+
+- localStorage
```

Comment:
IndexedDB for good measure?

Reply:
they can access the async api, just not the sync api. though I think that may have been deprecated.